### PR TITLE
Order of Loop - agent run first then tower tick().

### PR DESCRIPTION
**Pull request by udai-arneja**
Co-Authored-By: Priya Chhaya <77248784+priyachh@users.noreply.github.com>

# Summary

First the agent Run then the tower ticks. This makes more sense so the reshuffling doesn't happen before the agents have their last run.

## Additional Information

Also changed around logs a little so its gives better info.

## Test Plan

From the logs, they are more coherent. Day and tower resets occur at the right times.

### CODE REVIEW

**Comment by mdubied** [comment]
1) Current Simulation Tick should appear in the first position
2) Printing a Current Simulation Day would be nice to have

**Comment by mdubied** [comment]
to improve readibility:

- [ ] add a blank line or just a line (----------------------) at the end of every day

- [ ] only display "msg":"Reporting platform status end of day" at the end of the day and not at every tick

File: pkg/infra/tower.go [hunk 1]
```diff
@@ -46,9 +50,9 @@ func NewTower(maxPlatFood float64, agentCount,
 
 func (t *Tower) Tick() {
 	//logs
-	t.Log("", Fields{"Curr Tower Tick": t.dayInfo.CurrTick})
-	t.Log("Reporting platform status", Fields{"food_left": t.currPlatFood, "floor": t.currPlatFloor})
-
+	t.mx.RLock()
+	t.TowerStateLog(" end of day")
+	t.mx.RUnlock()
```
Comment (jzzheng22) [inline pkg/infra/tower.go:55]: I'm not sure we need this mutex - I don't think Tick() is ever called by multiple threads?

File: pkg/simulation/simAgent.go [hunk 2]
```diff
@@ -50,6 +49,7 @@ func (sE *SimEnv) replaceAgents(t *infra.Tower) {
 			sE.createNewAgent(t, agentType, floor)
 		}
 	}
+
```
Comment (jzzheng22) [inline pkg/simulation/simAgent.go:52]:
```suggestion
```

**Review by jzzheng22 — CHANGES_REQUESTED** [review]
Couple of questions but otherwise it's good, thanks

File: pkg/infra/tower.go (hunk 1 above)
Comment (udai-arneja) [reply pkg/infra/tower.go:55]: Truuuuee

File: pkg/infra/tower.go [hunk 3]
```diff
@@ -123,6 +127,7 @@ func (t *Tower) SendMessage(direction int, senderFloor int, msg messages.Message
 func (t *Tower) ResetTower() {
 	t.currPlatFood = t.maxPlatFood
 	t.currPlatFloor = 1
+	t.Log("Tower Reset", Fields{"currFood": t.currPlatFood, "maxFood": t.maxPlatFood, "currFloor": t.currPlatFloor})
```
[the remark this answers is not in the document]
Comment (udai-arneja) [reply pkg/infra/tower.go:130]: True guess we dont need to print it but its worth having a log saying reset imo?

[the remark this answers is not in the document]
Comment (jzzheng22) [reply pkg/infra/tower.go]: Yup, happy to print the reset, but don't need to pollute the logs with the rest of the info

File: pkg/infra/tower.go [hunk 4]
```diff
@@ -64,6 +66,7 @@ func (t *Tower) Tick() {
 	if t.dayInfo.CurrTick%t.dayInfo.TicksPerDay == 0 {
 		t.hpDecay() // decreases HP and kills if < 0
 		t.ResetTower()
+		t.Log("-----------------E-N-D----O-F----D-A-Y-----------------", Fields{})
```
Comment (jzzheng22) [inline pkg/infra/tower.go:69]:
```suggestion
		t.Log("-----------------END----OF----DAY-----------------", Fields{})
```

File: pkg/infra/tower.go [hunk 5]
```diff
@@ -46,9 +50,7 @@ func NewTower(maxPlatFood float64, agentCount,
 
 func (t *Tower) Tick() {
 	//logs
```
Comment (jzzheng22) [inline pkg/infra/tower.go:52]:
```suggestion
```

Comment (jzzheng22) [reply pkg/infra/tower.go:52]: The fact that you're calling the function `TowerStateLog()` should hopefully make it pretty clear that it is a log

**Review by jzzheng22 — CHANGES_REQUESTED** [review]

**Review by mdubied — CHANGES_REQUESTED** [review]
See my comments above

**Comment by udai-arneja** [comment]
@mdubied I think having it at the start and end of every tick is better for individual log reading?

**Review by mdubied — DISMISSED** [review]

**Comment by mdubied** [comment]
@udai-arneja the line you mean? I would keep it at the end of the day for now. Or single line for End of the tick, double line for end of the day

**Comment by udai-arneja** [comment]
@mdubied so keep it as it is. there is a. platform status before and after each tick and then a dashed line at the end of each day.

**Review by jzzheng22 — APPROVED** [review]

**Review by jzzheng22 — APPROVED** [review]

**Pull request merged**